pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -96,37 +96,50 @@ func machinesEqual(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
 	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
 }
 
+// isNodeJoin determines if a machine, in scope, should join of the cluster.
 func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachines []*clusterv1.Machine) (bool, error) {
 	switch set := scope.Machine.ObjectMeta.Labels["set"]; set {
 	case "node":
+		// Worker machines, not part of the controlplane, will always join the cluster.
 		return true, nil
 	case "controlplane":
+		// Controlplane machines will join the cluster if the cluster has an existing control plane.
+		controlplaneExists := false
+		var err error
 		for _, cm := range controlPlaneMachines {
```

Comment:
Should there be some type of locking here to avoid parallel initialization between multiple control plane hosts?

Reply:
Yes! But the upstream cluster deployer doesn't support multiple controlplane nodes in the namespace.
https://github.com/kubernetes-sigs/cluster-api/blob/master/cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go#L1045
So there will be no racing to be the first `controlplane` machine.
Once the upstream master start supporting multiple controlplane machines in the cluster, we'll have to make `isNodeJoin` thread safe.

Maybe worth adding comments about this.

Reply:
I created this issue https://github.com/kubernetes-sigs/cluster-api/issues/925 and I'll link it in the code.

Reply:
So, multiple control-plane nodes should work today if using `clusterctl` it will instantiate them serially. We even have a make target for easy testing `make create-cluster-ha`

Reply:
The serial instantiation is more for the benefit of `kubeadm` which will not support parallel `init` and controlplane `join` until v1.15.

Reply:
Correct. I had forgotten about that. Thanks for reminding.

Reply:
@detiber This PR, this is a bug fix. We can make `isNodeJoin` thread safe as an enhancement and fix to this issue https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/763 
I don't think it should block this PR, do you disagree?